src/common/veteran.js

```diff
@@ -642,4 +643,68 @@ const completeVeteran = {
   campLejeune: true
 };
 
-module.exports = { blankVeteran, completeVeteran };
+function veteranToApplication(veteran) {
+  // TODO(awong): Figure out how to do this w/o going through JSON.
+  return JSON.stringify(veteran, (key, value) => {
+    switch (key) {
+      // Convert radio buttons into booleans.
+      case 'isVaServiceConnected':
+      case 'compensableVaServiceConnected':
+      case 'provideSupportLastYear':
+      case 'receivesVaPension':
+      case 'provideFinancialInfo':
+      case 'understandsFinancialDisclosure':
+      case 'sameAddress':
+      case 'cohabitedLastYear':
+      case 'isCoveredByHealthInsurance':
+      case 'isMedicaidEligible':
+      case 'isEnrolledMedicarePartA':
+        return value.value === 'Y';
+
+      case 'veteranGrossIncome':
+      case 'veteranNetIncome':
+      case 'veteranOtherIncome':
+      case 'spouseGrossIncome':
+      case 'spouseNetIncome':
+      case 'spouseOtherIncome':
+      case 'grossIncome':
+      case 'netIncome':
+      case 'otherIncome':
+      case 'deductibleEducationExpenses':
+      case 'deductibleFuneralExpenses':
+      case 'deductibleMedicalExpenses':
+        return Number(value.value);
+
+      default:
+        // fall through.
+    }
+
+    // Turn date fields into ISO8601 dates.
+    if (value.day !== undefined && value.month !== undefined && value.year !== undefined) {
```

Comment:
we should use moment if it is still in the file or readd : http://momentjs.com/docs/#/displaying/as-iso-string/